### PR TITLE
Mizar setup and deployment scripts for Kubernetes.

### PR DESCRIPTION
**Pull request by phudtran**
etc: Autodeploy scripts  #24 

This PR does the following:
Adds scripts and daemonset yamls for setting up and running the transit daemon on a Kubernetes cluster.

### CODE REVIEW

File: etc/k8s/cluster_init.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function run_job() {
+    running="running"
+    done="done"
+
+    kubectl apply -f $1
+    validate $running $1 $2 $3
+    validate $done $1 $2 $3
+    kubectl delete -f $1
```
Comment (zasherif) [inline etc/k8s/cluster_init.sh]: Let's assign $1, $2, $3 to variables for readability

File: etc/k8s/cluster_init.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function validate() {
+    PODS=$(kubectl get pods | grep $3 | awk '{print $1}')
```
Comment (zasherif) [inline etc/k8s/cluster_init.sh]: ditto: assign ${number} to a variable

File: etc/k8s/images/kernelpatch-node.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM debian
```
Comment (zasherif) [inline etc/k8s/images/kernelpatch-node.Dockerfile]: Let's consolidate all dockerfiles in one file to create a single image. Move the runtime command to the daemon set..

File: etc/k8s/images/scripts/transitd-start.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+nsenter -t 1 -m -u -n -i ip link set dev eth0 xdpgeneric off # unload any existing xdp program
+nsenter -t 1 -m -u -n -i /trn_bin/transitd
```
Comment (zasherif) [inline etc/k8s/images/scripts/transitd-start.sh]: newline

File: kernelpatch.sh [hunk 5]
```diff
@@ -3,7 +3,7 @@
 kernel_dir="../mizar-kernel"
 mkdir -p $kernel_dir
 
-wget https://mizar.s3.amazonaws.com/kernel/linux-headers-5.2.0-mizar_5.2.0-mizar-2_amd64.deb -P kernel_dir
+wget https://mizar.s3.amazonaws.com/kernel/linux-headers-5.2.0-mizar_5.2.0-mizar-2_amd64.deb -P $kernel_dir
```
Comment (zasherif) [inline kernelpatch.sh]: Thanks for fixing this :)

File: etc/k8s/cluster_init.sh (hunk 1 above)
Comment (phudtran) [reply etc/k8s/cluster_init.sh]: Fixed in the latest commit!

File: etc/k8s/cluster_init.sh (hunk 2 above)
Comment (phudtran) [reply etc/k8s/cluster_init.sh]: Fixed in the latest commit!

File: etc/k8s/images/kernelpatch-node.Dockerfile (hunk 3 above)
Comment (phudtran) [reply etc/k8s/images/kernelpatch-node.Dockerfile]: Great idea! Fixed in the latest commit.

File: etc/k8s/images/scripts/transitd-start.sh (hunk 4 above)
Comment (phudtran) [reply etc/k8s/images/scripts/transitd-start.sh]: Fixed.

File: etc/k8s/kernelpatch-node-daemonset.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kernelpatch-node
+  namespace: default
```
Comment (vinaykul) [inline etc/k8s/kernelpatch-node-daemonset.yaml]: Please use kube-system namespace

File: etc/k8s/kernelpatch-node-daemonset.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
```
Comment (vinaykul) [inline etc/k8s/kernelpatch-node-daemonset.yaml]: Let's make this a init-container. See kube-flannel.yaml example

File: etc/k8s/node-init-daemonset.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
```
Comment (vinaykul) [inline etc/k8s/node-init-daemonset.yaml]: please make this also a init-container.

**Pull request merged**